gcc/rust/ast/rust-ast.h

```diff
@@ -1868,6 +1869,10 @@ class ASTFragment
    */
 
   bool is_single_fragment () const { return nodes.size () == 1; }
+  bool is_single_fragment (SingleASTNode::NodeType expected) const
```

Comment:
Not sure about the name of this method. Its a pet nit of mine that I prefer to use overloading only on methods where they already take arguments. It feels like a new method to check if is_single_fragment and that single fragment is of a particular type.

How about `is_single_fragment_of_kind (...)`

Reply:
Ah, I see what you mean. I'll rename it. `assert_single_fragment` kinda serves the same purpose but it also crashes if the assertion fails, so it's a bit different from `is_single_fragment`